ledger/acctonline.go

```diff
@@ -902,6 +902,13 @@ func (ao *onlineAccounts) TopOnlineAccounts(rnd basics.Round, voteRnd basics.Rou
 			if ot.Overflowed {
 				return nil, basics.MicroAlgos{}, fmt.Errorf("TopOnlineAccounts: overflow in stakeOfflineInVoteRound")
 			}
+			if params.StateProofExcludeTotalWeightWithRewards {
+				rewards := basics.PendingRewards(&ot, *params, oa.MicroAlgos, oa.RewardsBase, rewardsLevel)
+				totalOnlineStake = ot.SubA(totalOnlineStake, rewards)
```

Comment:
I don't think this takes into account that some accounts are non-participating, so they have no rewards. Despite the name, you can be "non-participating" which refers to earning rewards, but still registered to vote (I think).

So this would calculate an amount of rewards they _would_ earn, and subtract it. But actually, they will not earn it.

And does the original `totalOnlineStake` value take this distinction into account?

Reply:
https://github.com/algorand/go-algorand/blob/master/data/basics/userBalance.go#L35-L43 you can't be both NotParticipating and Online at the same time — Online implies rewards.

Reply:
That's to @cee for pointing out:
```
    // NotParticipating indicates that the associated account neither participates in the consensus, nor receives rewards.
```
I was mistaken.